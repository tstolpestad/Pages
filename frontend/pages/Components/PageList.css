.title{
	border-radius: 5px;
	background: lightblue;
	border: 1px solid cornflowerblue;
	padding: 5px;
	padding-left: 10px;
}
.rightIcon{
	float: right;
}
.indented {
	padding-left: 20px;
}
.list {

}
