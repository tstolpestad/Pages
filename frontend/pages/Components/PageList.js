//todo: add styling
//todo: optional header
//todo: make droppable
import React,{ useState } from "react";
import classNames from './PageList.css';
import {PageElement} from './PageElement';

export const PageList = ({pages, title})=>{

	const [open, setOpen] = useState(true);

	return <div>
		{title && <div className={classNames.title} onClick={()=> setOpen(!open)}>{title}</div>}
		{open && pages.map((page, index) => <PageElement key={index} text={page}/>)}
	</div>
}
