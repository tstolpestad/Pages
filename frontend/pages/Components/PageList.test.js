import {PageList} from './PageList'
import React from 'react';
import TestRenderer from 'react-test-renderer';

//this test has to use react's test renderer as enzyme does not have support for hooks yet
//Todo: consider changing to react-test-renderer for the entire app


describe("PageList", ()=> {
	const pages = ["test", "test/page"];
	const wrapper = TestRenderer.create(<PageList pages={pages} />);
	const wrapperWithTitle = TestRenderer.create(<PageList pages={pages} title="PageListTest"/>);
	it("should match snapshot for regression testing", ()=>{
		expect(wrapper.toJSON).toMatchSnapshot();
		expect(wrapperWithTitle.toJSON).toMatchSnapshot();
	});
	it("should render a list of page elements ", ()=> {
		console.log(wrapper.toJSON)
		expect(wrapper.root.findByType("PageElement").length).toEqual(2);
	});
	it("should have an optional title", ()=>{
		expect(wrapper.root.findAllByProps({className: "title"})).toHaveLength(0);
		expect(wrapperWithTitle.root.findAllByProps({className: "title"})).toHaveLength(1);
	});
	it("should be a droppable", ()=> {
		expect(false).toBe(true) //Todo:implement test
	});
	it("should be collapsable if it has a title", ()=>{
		expect(false).toBe(true) //Todo:implement test
	});
});
