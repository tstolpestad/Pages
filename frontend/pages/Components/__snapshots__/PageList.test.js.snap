// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageList should match snapshot for regression testing 1`] = `[Function]`;

exports[`PageList should match snapshot for regression testing 2`] = `[Function]`;
